### PR TITLE
Fisehara/translation tests

### PR DESCRIPTION
**Pull request by fisehara**

### CODE REVIEW

File: test/lib/test-init.ts [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
 ): Promise<ChildProcess> {
 	try {
-		const processArgs = {
-			fixturePath,
-			testListenPort: pineListenPort,
-			deleteDb: deleteDb || process.env.DELETE_DB,
+		// manually setting defaults
+		if (options.configPath === undefined) {
+			throw new Error(`config path not set`);
+		}
```
Comment (Page-) [inline test/lib/test-init.ts:15]: If you use
```ts
options: RequiredField<Partial<PineTestOptions>, 'configPath'>,
```
above then you can remove the need for this check (or at least make it able to be caught by typescript)
```suggestion
```
You can also use
```ts
OptionalField<PineTestOptions, 'listenPort' | 'deleteDb'>
```
which might be more accurate to your intentions

File: test/lib/test-init.ts [hunk 2]
```diff
@@ -1,19 +1,24 @@
 import { ChildProcess, fork } from 'child_process';
-export const testListenPort = 1337;
-export const testLocalServer = `http://localhost:${testListenPort}`;
+import { boolVar } from '@balena/env-parsing';
+import type { PineTestOptions } from './pine-init';
+export const listenPortDefault = 1337;
+export const testLocalServer = `http://localhost:${listenPortDefault}`;
 
+console.log(`listenPortDefault:${JSON.stringify(listenPortDefault, null, 2)}`);
```
Comment (Page-) [inline test/lib/test-init.ts:7]: Why are you stringifying a number here? Is this leftover debugging?

File: test/04-translations.test.ts [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+				resource: 'faculty',
+				body: {
+					name: faker.random.words(2),
+				},
+			})
+			.expect(201));
+	});
+
+	after(async () => {
+		await testDeInit(pineServer);
```
Comment (joshbwlng) [inline test/04-translations.test.ts:39]:
```suggestion
	after(() => {
		testDeInit(pineServer);
```

Pretty sure we don't need `await`s for `testDeInit()` calls.

File: test/lib/pine-in-process.ts [hunk 4]
```diff
@@ -1,22 +1,25 @@
...
-			fixturePath: string;
-			testListenPort: number;
-			deleteDb: boolean;
-		} = JSON.parse(process.argv[2]);
-		const initConfig = await import(processArgs.fixturePath);
-		console.info(`pine testListenPort: ${processArgs.testListenPort}`);
+		const processArgs: PineTestOptions = JSON.parse(process.argv[2]);
+		const initConfig = await import(processArgs.configPath);
+		console.info(`listenPort: ${processArgs.listenPort}`);
+		console.log(`processArgs:${JSON.stringify(processArgs, null, 2)}`);
```
[the remark this answers is not in the document]
Comment (fisehara) [reply test/lib/pine-in-process.ts:11]: Yes... thanks!

**Review by Page- — APPROVED** [review]

**Pull request merged**